### PR TITLE
Group 2 initial submission

### PR DESCRIPTION
**Pull request by pkkondamuri**
Pradeep 
Justin
Charlie

### CODE REVIEW

File: linked_list/implementation.py [hunk 1]
```diff
@@ -1,40 +1,108 @@
...
+            pop_elem = item.elem
+            prev_item.next = None
+            return pop_elem
+        else:
+            prev_item = self.__getitem__(index-1)
+            item = self.__getitem__(index)
+            pop_elem = item.elem
+            prev_item.next = item.next
+            return pop_elem
+
```
Comment (aroni11) [inline linked_list/implementation.py:105]: Great idea using **getitem**, wish I thought of that myself.

File: linked_list/implementation.py [hunk 2]
```diff
@@ -1,40 +1,108 @@
...
     def __eq__(self, other):
-        pass
-
+        
+        if len(self) != len(other):
+            return False
+        for elem1 in zip(self,other):
+            if elem1[0].elem != elem1[1].elem:
+                return False
+        return True
```
Comment (nsbrsarma) [inline linked_list/implementation.py:66]: This is very compact!!!

File: linked_list/implementation.py [hunk 3]
```diff
@@ -1,40 +1,108 @@
...
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+    
+        if elements:
+            for elem in elements:
+                self.append(elem)
+
```
Comment (aroni11) [inline linked_list/implementation.py:16]: Much simpler than our implementation, I like this one better.

File: linked_list/implementation.py [hunk 4]
```diff
@@ -1,40 +1,108 @@
...
+            return
+            
+        next_node = Node(elem)
+        self.end.next = next_node
+        self.end = next_node
 
     def count(self):
-        pass
+        
+        return self.__len__()
```
Comment (aroni11) [inline linked_list/implementation.py:82]: We went the other way around, calling count() in the **len** function. Don't know if it matters.

File: linked_list/implementation.py [hunk 5]
```diff
@@ -1,40 +1,108 @@
...
+            list_str += [self_start.elem]
+            self_start = self_start.next
+        return str(list_str)
 
     def __len__(self):
-        pass
+        length = 0
+        for elem in self:
+            length +=1
+        return length
```
Comment (aroni11) [inline linked_list/implementation.py:30]: Much better way than how we solved this, nice job.

**Pull request closed without merging**